### ISSUE TITLE
Apply circ_max_megabytes only to onion-related traffic

### ISSUE THREAD

**Issue by mikeperry-tor** [mikeperry-tor/vanguards#78]
From https://gitlab.torproject.org/tpo/core/tor/-/issues/40402:

I tried creating github account but I got instantly banned by github, so I wrote this here.

github.com/mikeperry-tor/vanguards/blob/master/README_TECHNICAL.md - Total Circuit Megabytes

My onion service server is completely torified (all connction routes to Tor) and there is a problem while using Tor with Vanguards.
If I set circ_max_megabytes to 30MB, my website visitors can interact with site just fine. However when I use Tor to download a-little-bit large file, like pulling something 100MB from torsocksd git, vanguards closed client connection.
Could you please separate client traffic and server traffic and apply circ_max_megabytes  to server traffic only?
TL:DR;
circ_max_megabytes should apply to onion service traffic (coming to onion & replying to onion customer) and not apply to localhost traffic (like torsocks git xyz)
